internal/executor/util/pod_util_test.go

```diff
@@ -308,6 +308,60 @@ func makePodsWithJobIds(jobIds []string) []*v1.Pod {
 	return pods
 }
 
+func TestExtractJobRunIds(t *testing.T) {
```

Comment:
I think these individual test cases could be refactored to be a table driven test.  

I find them nice because you write the boilerplate code once and then give multiple cases.